lib/recording/RecordingManager.js

```diff
@@ -239,14 +240,18 @@ class RecordingManager {
 
         }
 
+        function isRecordingFinish(element) {
+            return now - element.lastTimeUpdateChunk > element.recordingSessionDuration || element.playlistGenerator.getNumOfClips()
```

Comment:
1.now is used only here... maybe it's better to move the let now here..
2.arn't you missing element.playlistGenerator.getNumOfClips()   > XXX?